server/taskWorker.go

```diff
@@ -0,0 +1,298 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/trackit/jsonlog"
+	"github.com/trackit/trackit/es"
+
+	"github.com/trackit/trackit/awsSession"
+	"github.com/trackit/trackit/config"
+)
+
+type (
+	MessageData struct {
+		TaskName   string   `json:"task_name"`
+		Parameters []string `json:"parameters"`
+		LogStream  string   `json:"cloudwatch_log_stream"`
+	}
+
+	LogInput struct {
+		Message   string
+		Timestamp int64
+	}
+)
+
+const visibilityTimeoutInMinutes = 30
+const waitTimeInSeconds = 20
+const esHealthcheckTimeoutInMinutes = 10
+
+func taskWorker(ctx context.Context) error {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	logger.Info("Running task 'worker'.", nil)
+	sqsq := sqs.New(awsSession.Session)
+	cwl := cloudwatchlogs.New(awsSession.Session)
+
+	logger.Info("Retrieving SQS Queue URL.", nil)
+	queueUrl, err := retrieveQueueUrl(ctx, sqsq)
+
+	if err != nil {
+		return err
+	}
+
+	logger.Info("Waiting for messages.", nil)
+
+	var logsBuffer bytes.Buffer
+	writer := io.MultiWriter(os.Stdout, &logsBuffer)
+	ctx = jsonlog.ContextWithLogger(ctx, logger.WithWriter(writer))
+	logger = jsonlog.LoggerFromContextOrDefault(ctx)
+
+	for {
+		message, receiptHandle, err := getNextMessage(ctx, sqsq, queueUrl)
+		if err != nil {
+			logsBuffer.Reset()
+			continue
+		}
+
+		logger.Info("Received message, checking ES health and executing task.", map[string]interface{}{
+			"message": message,
+		})
+
+		if res, err := es.Client.ClusterHealth().Timeout(strconv.Itoa(esHealthcheckTimeoutInMinutes) + "m").Do(ctx); err != nil || res.TimedOut {
+			logger.Error("ES is not reachable.", map[string]interface{}{
+				"timeout": strconv.Itoa(esHealthcheckTimeoutInMinutes) + "m",
```

Comment:
`strconv.Itoa(esHealthcheckTimeoutInMinutes) + "m"` could be in a var as it is used multiple times